hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/MoveAssignmentList.java

```diff
@@ -37,6 +38,15 @@ public boolean add(JobTaskMoveAssignment assignment) {
         return super.add(assignment);
     }
 
+    public boolean contains(JobKey jobKey) {
```

Comment:
Change to `public boolean containsJob(String jobId)`

I also suggest `MoveAssignmentList` should not extend `ArrayList`. Instead it should have an internal `List` that can be accessed via a `getList` method, and expose the few List methods, such as `add`, `size` etc. that other code needs. In its current form, developers can be misled to make the wrong assumption about its behavior of methods such as `addAll`